api/hasura/actions/_handlers/deleteCircle.ts

```diff
@@ -0,0 +1,64 @@
+import type { VercelRequest, VercelResponse } from '@vercel/node';
+
+import { authCircleAdminMiddleware } from '../../../../api-lib/circleAdmin';
+import { adminClient } from '../../../../api-lib/gql/adminClient';
+import { errorResponseWithStatusCode } from '../../../../api-lib/HttpError';
+import {
+  deleteCircleInput,
+  composeHasuraActionRequestBody,
+} from '../../../../src/lib/zod';
+
+async function handler(req: VercelRequest, res: VercelResponse) {
+  const {
+    input: { payload },
+  } = composeHasuraActionRequestBody(deleteCircleInput).parse(req.body);
+
+  const { circle_id } = payload;
+
+  const {
+    circles: [existingCircle],
+  } = await adminClient.query(
+    {
+      circles: [
```

Comment:
it's a bit simpler and cleaner, i think, to use `circles_by_pk` here and then throw the 422 error if `deleted_at` isn't blank:

```
circles_by_pk: [
  { id: circle_id },
  { deleted_at: true }
]
```